catch by constant reference

## Description

In 

https://github.com/stan-dev/math/blob/develop/stan/math/prim/core/init_threadpool_tbb.hpp#L50

we `catch (boost::bad_lexical_cast)`, which throws a warning with `g++-8` and above saying

> stan/math/prim/core/init_threadpool_tbb.hpp:52:21: warning: catching polymorphic type ‘class boost::bad_lexical_cast’ by value [-Wcatch-value=]

The internet thinks that exceptions should be caught by constant reference.

## Example

Compile anything that includes this header file with `g++-8`

## Expected Output

None

#### Current Version:
v3.0.0

--- a/stan/math/prim/core/init_threadpool_tbb.hpp
+++ b/stan/math/prim/core/init_threadpool_tbb.hpp
@@ -47,7 +47,7 @@ inline int get_num_threads() {
                          "The STAN_NUM_THREADS environment variable is '",
                          "' but it must be positive or -1");
       }
-    } catch (boost::bad_lexical_cast) {
+    } catch (const boost::bad_lexical_cast&) {
       invalid_argument("get_num_threads(int)", "STAN_NUM_THREADS",
                        env_stan_num_threads,
                        "The STAN_NUM_THREADS environment variable is '",

